pkg/storage/stores/tsdb/single_file_index_test.go

```diff
@@ -372,3 +374,61 @@ func TestTSDBIndex_Stats(t *testing.T) {
 		})
 	}
 }
+
+func TestTSDBIndex_LabelVolume(t *testing.T) {
```

Comment:
I think we need some more tests here. Looks like we're only testing the all case, I think we should test matching subsets, maybe matching nothing, and filtering if we're going to keep it included.